spec/text-editor-component-spec.coffee

```diff
@@ -371,7 +407,7 @@ describe "TextEditorComponent", ->
       it "displays newlines as their own token outside of the other tokens' scopeDescriptor", ->
         editor.setText "var\n"
         nextAnimationFrame()
-        expect(component.lineNodeForScreenRow(0).innerHTML).toBe "<span class=\"source js\"><span class=\"storage modifier js\">var</span></span><span class=\"invisible-character\">#{invisibles.eol}</span>"
+        expect(component.lineNodeForScreenRow(0).innerHTML).toBe "<span class=\"source js\"><span class=\"storage modifier js\"><span data-start=\"0\" data-end=\"3\" class=\"token\">var</span></span></span><span class=\"invisible-character\">E</span>"
```

Comment:
I'm thinking we should avoid annotating the start/end positions on lines outside of the measurement context, since it adds complexity that isn't needed in the actual display lines. You could parameterize the line construction routine to only include it in certain cases.